$NetBSD: patch-build.conf,v 1.1 2013/09/02 19:08:24 thomasklausner Exp $

Add -fPIC to fix compilation on NetBSD/amd64.
Do not build OC TCAP and SCTP libraries (need extra libraries/kernel support).

--- build.conf.orig	2010-10-27 12:30:16.000000000 +0000
+++ build.conf
@@ -89,7 +89,7 @@ function check_tls_external_lib {
 # Linux executable generation definitions
 BUILD_EXE_CC_LINUX=g++
 #BUILD_EXE_CC_FLAGS_LINUX="-g -D_GNU_SOURCE -D_REENTRANT -Wall -Werror -DINIT_CALL_FILTER -DDEBUG_NEW -DLOG_DEBUG_NEW"
-BUILD_EXE_CC_FLAGS_LINUX="-DVERSION=\\\"${PROJECT_VERSION}\\\" -D_GNU_SOURCE -D_REENTRANT -Wall -Werror -DINIT_CALL_FILTER"
+BUILD_EXE_CC_FLAGS_LINUX="-DVERSION=\\\"${PROJECT_VERSION}\\\" -D_GNU_SOURCE -D_REENTRANT -Wall -Werror -DINIT_CALL_FILTER -fPIC"
 BUILD_EXE_CC_FLAGS_LINUX_DBG="-DDEBUG_MODE -g -D_GNU_SOURCE -D_REENTRANT -Wall -Werror -DINIT_CALL_FILTER -DLOG_DEBUG_NEW"
 BUILD_EXE_LD_LINUX=g++
 BUILD_EXE_LD_FLAGS_LINUX="-rdynamic"
@@ -102,7 +102,7 @@ BUILD_EXE_YACC_FLAGS_LINUX="-y -d"
 
 # Linux library generation definition
 BUILD_LIB_CC_LINUX=g++
-BUILD_LIB_CC_FLAGS_LINUX="-D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror"
+BUILD_LIB_CC_FLAGS_LINUX="-D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror -fPIC"
 BUILD_LIB_CC_FLAGS_LINUX_DBG="-g -DDEBUG_MODE -g -D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror -DLOG_DEBUG_NEW"
 BUILD_LIB_LD_LINUX=g++
 BUILD_LIB_LD_FLAGS_LINUX="-shared -fPIC"
@@ -121,28 +121,28 @@ fi
 
 # Linux library generation definition
 BUILD_LIBIPTLS_CC_LINUX=g++
-BUILD_LIBIPTLS_CC_FLAGS_LINUX="$OPENSSL_DIR_INC -DOPENSSL_NO_KRB5 -D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror"
+BUILD_LIBIPTLS_CC_FLAGS_LINUX="$OPENSSL_DIR_INC -DOPENSSL_NO_KRB5 -D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror -fPIC"
 BUILD_LIBIPTLS_LD_LINUX=g++
 BUILD_LIBIPTLS_LD_FLAGS_LINUX="-fPIC -shared"
 BUILD_LIBIPTLS_LD_LIBS_LINUX="-ldl -lpthread -lm $OPENSSL_DIR_LIB -lssl -lcrypto"
 
 # Linux library generation definition
 BUILD_LIBCRYPTO_CC_LINUX=g++
-BUILD_LIBCRYPTO_CC_FLAGS_LINUX="$OPENSSL_DIR_INC -DOPENSSL_NO_KRB5 -D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror"
+BUILD_LIBCRYPTO_CC_FLAGS_LINUX="$OPENSSL_DIR_INC -DOPENSSL_NO_KRB5 -D_GNU_SOURCE -D_REENTRANT -shared -Wall -Werror -fPIC"
 BUILD_LIBCRYPTO_LD_LINUX=g++
 BUILD_LIBCRYPTO_LD_FLAGS_LINUX="-fPIC -shared"
 BUILD_LIBCRYPTO_LD_LIBS_LINUX="-ldl -lpthread -lm $OPENSSL_DIR_LIB -lssl -lcrypto"
 
 # Linux library generation definition
 BUILD_LIBEXTSCTP_CC_LINUX=g++
-BUILD_LIBEXTSCTP_CC_FLAGS_LINUX="-I`check_sctp_external_lib`/include -D_GNU_SOURCE -D_REENTRANT -DSCTP_SOCKET_API -shared -Wall -Werror"
+BUILD_LIBEXTSCTP_CC_FLAGS_LINUX="-I`check_sctp_external_lib`/include -D_GNU_SOURCE -D_REENTRANT -DSCTP_SOCKET_API -shared -Wall -Werror -fPIC"
 BUILD_LIBEXTSCTP_LD_LINUX=g++
 BUILD_LIBEXTSCTP_LD_FLAGS_LINUX="-fPIC -shared"
 BUILD_LIBEXTSCTP_LD_LIBS_LINUX="-L`check_sctp_external_lib`/lib -lsctplib -lsctpsocket"
 
 # Linux library generation definition for using TCAP lib
 BUILD_TCAP_LIB_CC_LINUX=g++
-BUILD_TCAP_LIB_CC_FLAGS_LINUX="-D_GNU_SOURCE -DTCAP_API100 -D_REENTRANT -shared -Wall -Werror -I/opt/OC/include -I./octcap_include"
+BUILD_TCAP_LIB_CC_FLAGS_LINUX="-D_GNU_SOURCE -DTCAP_API100 -D_REENTRANT -shared -Wall -Werror -I/opt/OC/include -I./octcap_include -fPIC"
 BUILD_TCAP_LIB_LD_LINUX=g++
 BUILD_TCAP_LIB_LD_FLAGS_LINUX="-shared -fPIC"
 #BUILD_TCAP_LIB_LD_LIBS_LINUX="-lm -lntl"
@@ -289,14 +289,14 @@ BUILD_7_TYPE=lib
 BUILD_7_MODULES="common parser-frame filter-frame protocol-frame library-parser-msrp"
 
 # HP OC TCAP transport library
-BUILD_8_NAME="libtrans_octcap.so"
-BUILD_8_TYPE=tcap_lib
-BUILD_8_MODULES="common transport-frame protocol-frame library-trans-octcap32"
+#BUILD_8_NAME="libtrans_octcap.so"
+#BUILD_8_TYPE=tcap_lib
+#BUILD_8_MODULES="common transport-frame protocol-frame library-trans-octcap32"
 
 # sctp transport library
-BUILD_9_NAME="libtrans_extsctp.so"
-BUILD_9_TYPE=libextsctp
-BUILD_9_MODULES="common transport-frame protocol-frame library-trans-ip library-trans-extsctp"
+#BUILD_9_NAME="libtrans_extsctp.so"
+#BUILD_9_TYPE=libextsctp
+#BUILD_9_MODULES="common transport-frame protocol-frame library-trans-ip library-trans-extsctp"
 
 # tls transport library
 BUILD_10_NAME="libtrans_iptls.so"
